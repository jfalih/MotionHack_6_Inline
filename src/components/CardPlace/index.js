
import React from 'react'
import { TouchableOpacity, Image, View, Text } from 'react-native';

const CardPlace = ({title, source, onPress, description}) => {
    return(
        <TouchableOpacity onPress={onPress} style={{
            marginRight:15,
            width: 180,
            height: 220,
            borderRadius:20,
        }}>
            <Image resizeMode="cover" style={{
                width:180,
                height:220,
                borderRadius:20
            }} source={source}/>
            <View style={{
                width: 180,
                height: 220,
                padding:10,
                borderRadius:20,
                position: 'absolute',
                backgroundColor:'rgba(0,0,0,0.5)',
            }}>
                <View style={{
                    marginTop:'auto'
                }}>
                    <Text style={{
                        color:'#fff',
                        fontSize:18,
                        fontFamily:'Inter-Bold'
                    }}>{title}</Text>
                    <Text style={{
                        color:'#fff',
                        fontFamily:'Inter-Regular'
                    }}>{description}</Text>
                </View>
            </View>
        </TouchableOpacity>
    )
}

export default CardPlace