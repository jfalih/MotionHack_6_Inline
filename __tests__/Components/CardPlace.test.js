import React from 'react'
import renderer from 'react-test-renderer';
import CardPlace from './../../src/components/CardPlace/index';
it('renders correctly across screens', () => {
  const tree = renderer.create(<CardPlace />).toJSON();
  expect(tree).toMatchSnapshot();
});