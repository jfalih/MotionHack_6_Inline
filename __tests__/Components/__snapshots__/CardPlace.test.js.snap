// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 20,
      "height": 220,
      "marginRight": 15,
      "opacity": 1,
      "width": 180,
    }
  }
>
  <Image
    resizeMode="cover"
    style={
      Object {
        "borderRadius": 20,
        "height": 220,
        "width": 180,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "borderRadius": 20,
        "height": 220,
        "padding": 10,
        "position": "absolute",
        "width": 180,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": "auto",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Inter-Bold",
            "fontSize": 18,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Inter-Regular",
          }
        }
      />
    </View>
  </View>
</View>
`;
